truenas_install/__main__.py

```diff
@@ -465,14 +465,20 @@ def main():
                                                 "--efi-directory=/boot/efi",
                                                 "--bootloader-id=debian",
                                                 "--recheck",
-                                                "--no-floppy"]
-                                    if not os.path.exists("/sys/firmware/efi"):
-                                        grub_cmd.append("--no-nvram")
+                                                "--no-floppy",
+                                                "--no-nvram"]
                                     run_command(grub_cmd)
 
                                     run_command(["chroot", root, "mkdir", "-p", "/boot/efi/EFI/boot"])
                                     run_command(["chroot", root, "cp", "/boot/efi/EFI/debian/grubx64.efi",
                                                  "/boot/efi/EFI/boot/bootx64.efi"])
+
+                                    if os.path.exists("/sys/firmware/efi"):
+                                        run_command(["chroot", root, "efibootmgr", "-c",
+                                                     "-d", f"/dev/{disk}",
+                                                     "-p", "2",
```

Comment:
Is this EFI partition number? Above we have `efi_partition_number` which can also be `1` if we are upgrading a FreeBSD installation.

Reply:
Yes, this is EFI partition number. I will update accordingly